kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+int sem_destroy(sem_t *semaphore)
+{
+	k_sem_reset(semaphore->sem);
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	int count;
+
+	count = k_sem_count_get(semaphore->sem);
+	*value = count;
```

Comment:
count seems useless here.

Reply:
@tbursztyka  Here I am trying to write count to *value. and then user can read that pointer if semaphore count is needed.

Reply:
Sure, but count is variable is useless. Just do *value = k_sem_count_get(...)